File: dehaze-python/app/models/entity/sys_feedback_reply.py
from typing import Optional

from sqlalchemy import BigInteger, String, JSON, SmallInteger
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class SysFeedbackReply(BaseModel):
    __tablename__ = 'sys_feedback_reply'
    __table_args__ = {'comment': '反馈回复表'}

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True, comment='主键')
    feedback_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment='反馈ID')
    replier_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment='回复人ID')
    replier_type: Mapped[int] = mapped_column(mysql_types.TINYINT, nullable=False, comment='回复人类型(1:用户;2:管理员)')
    content: Mapped[str] = mapped_column(String(2000), nullable=False, comment='回复内容')
    reply_type: Mapped[Optional[str]] = mapped_column(String(32), nullable=True, comment='回复类型')
    attachments: Mapped[Optional[list]] = mapped_column(JSON, nullable=True, comment='附件URL（JSON数组）')
    deleted: Mapped[int] = mapped_column(SmallInteger, nullable=False, default=0, comment='逻辑删除标识(0:未删除;1:已删除)')

File: dehaze-go/internal/model/sys_feedback_reply.go
package model

import "time"

type SysFeedbackReply struct {
	ID          int64     `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	FeedbackID  int64     `gorm:"column:feedback_id;type:bigint;not null;index:idx_feedback_id_create_time,priority:1;comment:反馈ID" json:"feedbackId"`
	ReplierID   int64     `gorm:"column:replier_id;type:bigint;not null;index:idx_replier_id;comment:回复人ID" json:"replierId"`
	ReplierType int8      `gorm:"column:replier_type;type:tinyint;not null;comment:回复人类型(1:用户;2:管理员)" json:"replierType"`
	Content     string    `gorm:"column:content;type:varchar(2000);not null;comment:回复内容" json:"content"`
	ReplyType   string    `gorm:"column:reply_type;type:varchar(32);comment:回复类型(info:信息补充;resolved:已解决;unsupported:暂不支持;dev_transfer:转开发)" json:"replyType"`
	Attachments string    `gorm:"column:attachments;type:json;comment:附件URL（JSON数组）" json:"attachments"`
	Deleted     int8      `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识" json:"-"`
	CreateTime  time.Time `gorm:"column:create_time;type:datetime;default:CURRENT_TIMESTAMP;index:idx_feedback_id_create_time,priority:2;comment:创建时间" json:"createTime"`
	UpdateTime  time.Time `gorm:"column:update_time;type:datetime;default:CURRENT_TIMESTAMP;comment:更新时间" json:"updateTime"`
	CreateBy    *int64    `gorm:"column:create_by;comment:创建人ID" json:"createBy"`
	UpdateBy    *int64    `gorm:"column:update_by;comment:修改人ID" json:"updateBy"`
}

func (SysFeedbackReply) TableName() string {
	return "sys_feedback_reply"
}
